src/main/java/com/nutomic/syncthingandroid/activities/ShareActivity.java

```diff
@@ -52,12 +65,24 @@ public void onApiChange(SyncthingService.State currentState) {
 
         List<Folder> folders = getApi().getFolders();
 
+        // Get the index of the previously selected folder.
+        int folderIndex = 0;
+        String savedFolderId = getSharedPreferences(SHARED_PREFS_SAVED_DIRECTORIES, MODE_PRIVATE)
```

Comment:
In that case, should I use something like saved_sub_directory_(insert folder id here) as the key for the saved sub directories, instead of just the folder id. Just to ensure that the key is unique?